src/utils/validation/auth/addUserSchema.ts

```diff
@@ -0,0 +1,66 @@
+import Joi from 'joi';
+const addUserSchema = Joi.object({
+  email: Joi.string()
+    .email({minDomainSegments: 2,tlds: { allow: ['com', 'net'] },})
+    .required()
+    .messages({
+      'string.email': 'Email must be a valid email address',
+      'string.empty': 'Email cannot be empty',
+      'any.required': 'Email is required',
+    }),
+  password: Joi.string()
+    .min(8)
+    .required()
+    .pattern(
+      new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^\\da-zA-Z]).+$')
+    )
+    .messages({
+      'string.pattern.base':
+      'Password must include at least one lowercase letter, one uppercase letter, one number, and one special character',
+      'string.min': 'Password must be at least 8 characters long',
+      'string.empty': 'Password cannot be empty',
+      'any.required': 'Password is required',
+    }),
+  full_name: Joi.string()
+    .min(3)
+    .max(30)
+    .required()
+    .messages({
+      'string.empty': 'Full name cannot be empty',
+      'any.required': 'Full name is required',
+    }),
+  username: Joi.string()
+    .min(3)
+    .max(30)
+    .alphanum()
+    .lowercase()
+    .uppercase()
+    .required()
+    .messages({
+      'string.empty': 'Username cannot be empty',
+      'any.required': 'Username is required',
+    }),
+  cohort_id: Joi.number()
+    .integer()
+    .min(1)
+    .max(100)
```

Comment:
The `cohort_id` is a serial number from `postgres` so it will start from 0, not 1, and delete the `max` prop Coz maybe we have more than 100 roles or cohorts so delete it.